test/test-assertions.rb

```diff
@@ -1877,7 +1877,7 @@ def test_fail_with_float_like_object
         def object.to_f
           0.4
         end
-        inspected_object = AssertionMessage.convert(object)
+        AssertionMessage.convert(object)
```

Comment:
Could you use `inspected_object` in the `check_fail()` argument instead of removing `inspected_object`?